test/osb_and_plugin_test/osb_test/instance_sharing_test.go

```diff
@@ -0,0 +1,394 @@
+
+		When("Provision", func() {
+			Context("in CF platform", func() {
+				BeforeEach(func() {
+					platformJSON = common.MakePlatform("cf-platform", "cf-platform", "cloudfoundry", "test-platform-cf")
+				})
+				It("creates reference instance successfully", func() {
+					_, referenceInstanceID = executeProvisionTest(platform, false)
+				})
+				It("creates reference instance successfully and return operation when async=true", func() {
+					_, sharedInstanceID := createAndShareInstance(false)
+					resp, referenceInstanceID := createReferenceInstance(platform.ID, sharedInstanceID, false)
+
+					respObject := resp.JSON().Object()
+					fmt.Print(respObject)
+					obj := VerifyResourceExists(ctx.SMWithOAuthForTenant, ResourceExpectations{
+						ID:    referenceInstanceID,
+						Type:  types.ServiceInstanceType,
+						Ready: true,
+					})
+					obj.
+						ContainsKey("platform_id").
+						ValueEqual("platform_id", platform.ID)
+				})
+
+			})
+
+			Context("in K8S platform", func() {
+				BeforeEach(func() {
+					platformJSON = common.MakePlatform("k8s-platform", "k8s-platform", "kubernetes", "test-platform-k8s")
+				})
+				It("creates reference instance successfully", func() {
+					_, referenceInstanceID = executeProvisionTest(platform, false)
+				})
+			})
+		})
+
+		When("Deprovision", func() {
+
+			Context("in CF platform", func() {
+				BeforeEach(func() {
+					platformJSON = common.MakePlatform("cf-platform", "cf-platform", "cloudfoundry", "test-platform-cf")
+				})
+
+				AfterEach(func() {
+					VerifyResourceDoesNotExist(ctx.SMWithOAuthForTenant, ResourceExpectations{
+						ID:   referenceInstanceID,
+						Type: types.ServiceInstanceType,
+					})
+
+				})
+
+				It("deletes reference instance successfully", func() {
+					_, referenceInstanceID = executeProvisionTest(platform, false)
+					deleteInstance(referenceInstanceID)
+				})
+			})
+
+			Context("in K8S platform", func() {
+				BeforeEach(func() {
+					platformJSON = common.MakePlatform("k8s-platform", "k8s-platform", "kubernetes", "test-platform-k8s")
+				})
+
+				It("deletes reference instance successfully", func() {
+					_, referenceInstanceID = executeProvisionTest(platform, false)
+					resp := ctx.SMWithBasic.DELETE(instanceSharingBrokerURL+"/v2/service_instances/"+referenceInstanceID).
+						WithHeader(brokerAPIVersionHeaderKey, brokerAPIVersionHeaderValue).
+						WithQuery("async", "false").
+						Expect().Status(http.StatusOK)
+					fmt.Print(resp)
+					//deleteInstance(referenceInstanceID)
+				})
+
+			})
+		})
+
+		When("Bind", func() {
+
+			AfterEach(func() {
+			})
+			Context("in CF platform", func() {
+				var cfSharedInstanceID string
+				var k8sReferenceInstanceID string
+				BeforeEach(func() {
+					platformJSON = common.MakePlatform("cf-platform", "cf-platform", "cloudfoundry", "test-platform-cf")
+					instanceSharingBrokerServer.ShouldRecordRequests(true)
+				})
+				It("binds reference instance successfully", func() {
```

Comment:
add a test to validate that the context switch happens correctly (shared instance context , passed to broker)